galgebra/mv.py

```diff
@@ -191,69 +196,72 @@ def _make_grade(ga, *args, **kwargs):
         else:
             raise TypeError("Expected a string, list, or tuple")
 
     @staticmethod
-    def _make_scalar(ga, *args, **kwargs):
+    def _make_scalar(ga, __name_or_value, **kwargs):
         """ Make a scalar multivector """
-        if utils.isstr(args[0]):
+        if utils.isstr(__name_or_value):
+            name = __name_or_value
             if 'f' in kwargs and isinstance(kwargs['f'],bool):
                 if kwargs['f']:
-                    return Function(args[0])(*ga.coords)
+                    return Function(name)(*ga.coords)
                 else:
-                    return Symbol(args[0], real=True)
+                    return Symbol(name, real=True)
             else:
                 if 'f' in kwargs and isinstance(kwargs['f'],tuple):
-                    return Function(args[0])(*kwargs['f'])
+                    return Function(name)(*kwargs['f'])
         else:
-            return args[0]
+            value = __name_or_value
+            return value
 
     @staticmethod
-    def _make_vector(ga, *args, **kwargs):
+    def _make_vector(ga, __name_or_coeffs, **kwargs):
         """ Make a vector multivector """
-        return Mv._make_grade(ga, args[0], 1, **kwargs)
+        return Mv._make_grade(ga, __name_or_coeffs, 1, **kwargs)
 
     @staticmethod
-    def _make_bivector(ga, *args, **kwargs):
+    def _make_bivector(ga, __name_or_coeffs, **kwargs):
         """ Make a bivector multivector """
-        return Mv._make_grade(ga, args[0], 2, **kwargs)
+        return Mv._make_grade(ga, __name_or_coeffs, 2, **kwargs)
 
     @staticmethod
-    def _make_pseudo(ga, *args, **kwargs):
+    def _make_pseudo(ga, __name_or_coeffs, **kwargs):
         """ Make a pseudo scalar multivector """
-        return Mv._make_grade(ga, args[0], ga.n, **kwargs)
+        return Mv._make_grade(ga, __name_or_coeffs, ga.n, **kwargs)
 
     @staticmethod
-    def _make_mv(ga, *args, **kwargs):
+    def _make_mv(ga, __name, **kwargs):
         """ Make a general (2**n components) multivector """
-        tmp = Mv._make_scalar(ga, args[0], **kwargs)
+        tmp = Mv._make_scalar(ga, __name, **kwargs)
         for grade in ga.n_range:
-            tmp += Mv._make_grade(ga, args[0], grade + 1, **kwargs)
+            tmp += Mv._make_grade(ga, __name, grade + 1, **kwargs)
         return tmp
 
     @staticmethod
-    def _make_spinor(ga, *args, **kwargs):
+    def _make_spinor(ga, __name, **kwargs):
         """ Make a general even (spinor) multivector """
-        tmp = Mv._make_scalar(ga, args[0], **kwargs)
+        tmp = Mv._make_scalar(ga, __name, **kwargs)
         for grade in ga.n_range:
             if (grade + 1) % 2 == 0:
-                tmp += Mv._make_grade(ga, args[0], grade + 1, **kwargs)
+                tmp += Mv._make_grade(ga, __name, grade + 1, **kwargs)
         return tmp
 
     @staticmethod
-    def _make_odd(ga, *args, **kwargs):
+    def _make_odd(ga, __name_or_coeffs, **kwargs):
```

Comment:
Note that unlike `even`, `odd` accepts not only a name, but coeffs of length `l = 1 if ga.n % 2 == 1 else ga.n`, with semantics _"set the coeffiencients for the first `l` lexicographically sorted elements of each grade to `coeffs`"_. This is nonsense, but tracked in #81 to avoid scope creep.